internal/core/items.rs

```diff
@@ -200,6 +220,18 @@ impl ItemWeak {
     pub fn upgrade(&self) -> Option<ItemRc> {
         self.component.upgrade().map(|c| ItemRc::new(c, self.index))
     }
+
+    pub fn component(&self) -> ComponentWeak {
+        self.component.clone()
+    }
+
+    pub fn index(&self) -> usize {
+        self.index
+    }
+
+    pub fn is_none(&self) -> bool {
```

Comment:
Please provide good documentation that this can return false even if the upgrade would return None.

But as discussed in the other PR, you probably should consider using ItemRc or Option<ItemRc> instead.